site2/docs/functions-runtime.md

```diff
@@ -0,0 +1,148 @@
+---
+id: functions-runtime
+title: Configure Functions runtime
+sidebar_label: Configure Functions runtime
+---
+
+Currently, Pulsar Functions supports running functions in three runtimes.
+
+- *Thread*: Invoke functions in threads in Functions Worker.
+- *Process*: Invoke functions in processes forked by Functions Worker.
+- *Kubernetes*: Submit functions as Kubernetes StatefulSets by Functions Worker.
+
+## Configure Thread runtime
+
+*Thread* runtime is the most simplest runtime to configure. At most of the time, you don't need
+to configure anything. You can customize the thread group name by setting following settings:
+
+```yaml
+threadContainerFactory:
+  threadGroupName: "Your Function Container Group"
+```
+
+*Thread* runtime only supports Java language.
+
+## Configure Process runtime
+
+Similar as *Thread* runtime, at most of the time, you don't need to configure anything special when
+enabling *Process* runtime.
+
+```yaml
+processContainerFactory:
+  # the directory for storing the function logs
+  logDirectory:
+  # change the jar location only when you put the java instance jar in a different location
+  javaInstanceJarLocation:
+  # change the python instance location only when you put the python instance jar in a different location
+  pythonInstanceLocation:
+  # change the extra dependencies location:
+  extraFunctionDependenciesDir:
+```
+
+Currently Pulsar supports running Java, Python, and Go functions in Process runtime mode.
+
+## Configure Kubernetes runtime
+
+Configuring Kubernetes runtime is as simple as configuring the other two runtimes. You can just uncomment
```

Comment:
What's the meaning of "uncomment"?

Reply:
:) try google it. @Jennifer88huang 
```
https://www.google.com.hk/search?newwindow=1&safe=strict&ei=-9PYXJWHEPKwmAWSqZdQ&q=uncomment&oq=uncomment&gs_l=psy-ab.3..0i10i70i255j0i10l3j0i67j0l5.1725.1725..2054...0.0..0.104.104.0j1......0....1..gws-wiz.t1uIqcWY1uA
```

Reply:
@jiazhai Thank you very much for your information. 
When you ''uncomment" sth in code, you actually want to enable some configuration options; in some cases, you just disable it with "comment", that's why we use comment lines in code. Here I want to tell @sijie, he could use "enable" / "disable" or other similar words to replace "uncomment"/"comment" in technical writing. When he explains the meaning, he will realize it.

Reply:
@Jennifer88huang I think from technical writing perspective, "enabling" and "disabling" are probably better. However sometimes "enabling" and "disabling" are too general that users have no idea to enable or disable. Can we use a phrase like "enabling sth by uncommenting ..."?

Reply:
Sure, it's much better.